src/tree_cmd.cpp

```diff
@@ -99,11 +99,14 @@ static bool IsNearbyForest(TileIndex tile)
 	/* Count the trees around the clear tile to determine if it's near a forest */
 	for (TileIndex t : TileArea(tile).Expand(FOREST_SEARCH_RADIUS)) {
 		if (IsTileType(t, MP_TREES)) {
-			planted_tile_count++;
+			if (++planted_tile_count >= FOREST_THRESHOLD)
+			{
+				return true;
+			}
 		}
```

Comment:
```suggestion
		if (!IsTileType(t, MP_TREES)) continue;
		++planted_tile_count;
		if (planted_tile_count >= FOREST_THRESHOLD) return true;
```

1) early continue.
2) prefer to avoid side effects in if-conditions.
3) unnecessary bracing.